db/migrate/20221019091238_add_user_ref_to_post.rb

```diff
@@ -0,0 +1,5 @@
+class AddUserRefToPost < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: true
```

Comment:
- [ ] The posts table should have an `author_id`, not `user_id`, and so this reference should be `author`, not `user`. Kindly run the necessary migration to fix this. You can look at this [StackOverflow answer](https://stackoverflow.com/a/42056089) for ideas on how to specify references for different column names

Reply:
thanks for the review and the valuable resources you provided